packages/cozy-client/src/hooks/useClientErrors.spec.jsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react'
+import useClientErrors from './useClientErrors'
+import { renderHook, act } from '@testing-library/react-hooks'
+import CozyProvider from '../Provider'
+import { shallow } from 'enzyme'
+import { FetchError } from 'cozy-stack-client'
+
+function createCozyClient() {
+  return {
+    on: jest.fn(),
+    removeListener: jest.fn()
+  }
+}
+
+function createWrapper(client = createCozyClient()) {
+  function Wrapper({ children }) {
+    return <CozyProvider client={client}>{children}</CozyProvider>
+  }
+  return Wrapper
+}
+
+function renderWrappedHook(params, client) {
+  const wrapper = createWrapper(client)
+  return renderHook(() => useClientErrors(), { wrapper })
+}
+
+function wrappedShallow(tree, client) {
+  return shallow(tree, { wrappingComponent: createWrapper(client) })
+}
+
+describe('useClientErrors', () => {
+  it('registers an `error` handler in client', done => {
+    const client = createCozyClient()
+    client.on.mockImplementation(name => name === 'error' && done())
+    renderWrappedHook(undefined, client)
+  })
+
+  describe('renderErrors', () => {
+    it('returns a function', () => {
+      const { result } = renderWrappedHook()
+      const { ClientErrors } = result.current
+      expect(ClientErrors).toBeInstanceOf(Function)
+    })
```

Comment:
I am not sure this is necessary to test for this. We expect ClientErrors to be either a function or a React.Component derived class. This would be tested by the following tests when ClientErrors is mounted.